delivery/services/organise_service.py

```diff
@@ -207,3 +291,198 @@ def organise_sample_file(self, sample_file, organised_sample_path, lanes):
             read_no=sample_file.read_no,
             is_index=sample_file.is_index,
             checksum=sample_file.checksum)
+
+    @staticmethod
+    def load_yaml_config(config_file_path):
+        """
+        Open and read yaml configuration file for 
+        organising runfolders or projects.
+        :param config_file_path: /path/to/config.yaml
+        :return: Config as dict. 
+        """
+        with open(config_file_path) as f:
+            config_as_dict = yaml.load(f, Loader=yaml.SafeLoader)
+
+        return config_as_dict
+    
+    @staticmethod
+    def parse_yaml_config(config_file_path, input_value):
+        """
+        Read a configuration file for organizing a runfolder 
+        or analyzed project prior to delivery. 
+        This function will do string substitutions based on the input, 
+        keys in the configuration file and results from resolved 
+        regular expressions (based on patterns from the configuration file). 
+
+        :param config_file_path: Full path to a 
+        configuration file in yaml-format.
+        :param input_value: A runfolder name or a 
+        project name (for analysed projects).
+        :return: A list with one element per link/copy operation to be 
+        performed during organisation.
+         Each element consist of a tuple with 3 values 
+         ("path/to/source/file", 
+         "path/to/destination/file", 
+         "options as dictionary").
+        """
+
+        config = OrganiseService.load_yaml_config(config_file_path)
+
+        config["variables"]["inputkey"] = input_value
+        variables_dict = config["variables"]
+
+        config_as_list = []
+
+        for key in variables_dict.keys():
+            #Perform string substitutions on variable
+            #  values based on variable keys.
+            variables_dict[key] = OrganiseService.sub_values_with_dict(
+                variables_dict[key], 
+                variables_dict)
+
+        for file_element in config["files_to_organise"]:
+            # for each source-destination pair, 
+            # under "files_to_organise", given in the configuration file
+            config_as_list = config_as_list + OrganiseService.process_file_to_organise(
```

Comment:
the `+` operator will add all elements from the second list to the first. use `config_as_list.append()` instead